mod/settings.php

```diff
@@ -364,41 +364,44 @@ function settings_post(App $a) {
 
 	call_hooks('settings_post', $_POST);
 
-	if((x($_POST,'password')) || (x($_POST,'confirm'))) {
+	if ((x($_POST,'password')) || (x($_POST,'confirm'))) {
 
 		$newpass = $_POST['password'];
 		$confirm = $_POST['confirm'];
 		$oldpass = hash('whirlpool', $_POST['opassword']);
 
 		$err = false;
-		if($newpass != $confirm ) {
+		if ($newpass != $confirm ) {
 			notice( t('Passwords do not match. Password unchanged.') . EOL);
 			$err = true;
 		}
 
-		if((! x($newpass)) || (! x($confirm))) {
+		if ((! x($newpass)) || (! x($confirm))) {
 			notice( t('Empty passwords are not allowed. Password unchanged.') . EOL);
 			$err = true;
-        }
+		}
 
-        //  check if the old password was supplied correctly before
-        //  changing it to the new value
-        $r = q("SELECT `password` FROM `user`WHERE `uid` = %d LIMIT 1", intval(local_user()));
-        if( $oldpass != $r[0]['password'] ) {
-            notice( t('Wrong password.') . EOL);
-            $err = true;
-        }
+		//  check if the old password was supplied correctly before
+		//  changing it to the new value
+		$r = q("SELECT `password` FROM `user`WHERE `uid` = %d LIMIT 1", intval(local_user()));
+		if (!dbm::is_result($r)) {
+			killme();
```

Comment:
Will be included in next PR. I have had already added a `@TODO` so it won't be forgotten.